search.py

```diff
@@ -283,3 +316,51 @@ def search(input_root, data_tree):
                 # elements don't match->stop input iteration and move to next data element
                 break
     return measure_match_list
+
```

Comment:
All lint output for the lines below (l321-l366) needs to be fixed.

Reply:
are there any issues in particular that you feel need to be changed before you can accept?

Reply:
@justinnhli

Reply:
Not outside of what I commented on. You added test cases for the new functions, so that's good.